code/integration-tests/runtime-tests/tests/multihopTests.ts

```diff
@@ -0,0 +1,265 @@
+import {
+  centauriAddress,
+  composableEndpoint,
+  ksmOncentauri,
+  ksmOnComposable,
+  ksmOnOsmosis,
+  kusamaEndpoint,
+  osmosisAddress,
+  picassoEndpoint,
+  picassoEscrowAddress,
+  picassoFeeAddress,
+  polkadotEndpoint
+} from "./utils/constants";
+import {ApiPromise} from "@polkadot/api";
+import {Asset} from "./utils/types";
+import {waitForBlocks} from "./utils/txClient";
+import BigNumber from "bignumber.js";
+import {mintAssetsOnRelays, mintAssetsToWallets} from "./utils/mintingHelper";
+import {addFeelessChannels, waitForSeconds} from "./utils/ibcUtils";
+
+const chai = require('chai');
+const expect = require('chai').expect;
+chai.use(require('chai-bignumber')());
+BigNumber.config({ROUNDING_MODE: BigNumber.ROUND_DOWN})
+
+
+describe('MultiHop Tests', function () {
+  this.timeout(30 * 60 * 1000);
+  const picassoAssets = createAssets('picasso');
+  const composableAssets = createAssets('composable');
+  let centauriAssets = createAssets('centauri');
+  let osmosisAssets = createAssets('osmosis');
+  const ksmTransferAmount = new BigNumber('10000000000000');
+  const ksmOnKusama = createAsset("1", 12, 'kusama', true);
+  const dotOnPolkadot = createAsset("1", 10, 'composable', true);
+  let ibcSentAmount: BigNumber;
+  let sudoKey: KeyringPair;
+  let testWallet: KeyringPair;
+  let picassoApi: ApiPromise;
+  let composableApi: ApiPromise;
+  let polkadotApi: ApiPromise;
+  let kusamaApi: ApiPromise;
+  let ibcEvent: any;
+
+  before('Mint assets and add routes', async () => {
+    picassoApi = await initializeApi(picassoEndpoint);
+    composableApi = await initializeApi(composableEndpoint);
+    kusamaApi = await initializeApi(kusamaEndpoint);
+    polkadotApi = await initializeApi(polkadotEndpoint);
+    ({sudoKey, testWallet} = getWallets('multihop'));
+    await Promise.all([
+      mintAssetsToWallets(picassoApi, testWallet, sudoKey, [1, 4, 6, 130], '10000000000000000', 'picasso'),
+      mintAssetsOnRelays([kusamaApi, polkadotApi], sudoKey, testWallet.address),
+    ]);
+    await createRoute(picassoApi, sudoKey, 2, [['picasso', 'composable'], ['composable', 'picasso']]);
+    await createRoute(picassoApi, sudoKey, 3, [['picasso', 'centauri'], ['centauri', 'osmo']]);
+    await Promise.all([
+      getBalance(picassoAssets, [testWallet.address, picassoFeeAddress, picassoEscrowAddress], 'dotsama', picassoApi),
+      getBalance(composableAssets, [testWallet.address], 'dotsama', composableApi),
+      getBalance(ksmOnKusama, [testWallet.address], 'dotsama', kusamaApi),
+      getBalance(dotOnPolkadot, [testWallet.address], 'dotsama', polkadotApi),
+      getBalance(centauriAssets, [centauriAddress], 'cosmos'),
+      getBalance(osmosisAssets, [osmosisAddress], 'cosmos'),
+    ]);
+    await addFeelessChannels(composableApi, sudoKey);
+  })
+
+  before('Wait for channel 1 to open to centauri', async () => {
+    await waitForChannelsToOpen(2, 'centauri');
+  });
+
+  it('Starts a transfer to osmosis', async () => {
```

Comment:
please be most specific, like `transfer: picass->centauri->osmosis`

Reply:
done.